crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -502,6 +502,11 @@ where
                     .flatten();
                 let (from_block_number, to_block_number) =
                     logs_utils::get_filter_block_range(from, to, start_block, info);
+                // if function signaled empty range:
+                if from_block_number == 0 && to_block_number == 0 {
+                    return Ok(Vec::new()); // returns empty log set, not an error
+                }
```

Comment:
this isn't very intuitive that we now need to check for == 0

we can add the check directly to the variants from,to blocks